packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -266,7 +266,7 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
             isString = ClientModelPlugin.Instance.TypeFactory.CreateCSharpType(inputParam.Type)?.Equals(typeof(string)) == true;
             if (inputParam.Kind == InputOperationParameterKind.Constant)
             {
-                valueExpression = Literal(inputParam.DefaultValue?.Value);
+                valueExpression = Literal((inputParam.Type as InputLiteralType)?.Value);
```

Comment:
So what if the kind of this parameter is constant, but its type is not a literal?
Would this happen?
If this should never happen, we should add an assertion somewhere to ensure it because it looks like that it does not make sense if the kind is constant but the type is not a literal

Reply:
Looks like content type parameters will be always `constant` no matter the exact type: https://github.com/microsoft/typespec/blob/4ee42d91643c4090ff1eb032506f7e26884b0def/packages/http-client-csharp/emitter/src/lib/operation-converter.ts#L380-L381